File: cpp/129_SumRoottoLeafNumbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        if(root == NULL) return 0;
        return sumTree(root,0);
    }
private:
    int sumTree(TreeNode *root, int val){
        if(root->left == NULL && root->right == NULL) return val * 10 + root->val;
        int ret = 0;
        if(root->left != NULL){
            ret += sumTree(root->left, 10*val + root->val);
        }
        if(root->right != NULL){
            ret += sumTree(root->right, 10*val + root->val);
        }
        return ret;
    }
};

File: js/129_SumRoottoLeafNumbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    if(root === null) return 0;
    return sumTree(root,0);
};

var sumTree = function(root, val){
    if(root.left === null && root.right === null) return val*10 + root.val;
    var ret = 0;
    if(root.left !== null) ret += sumTree(root.left, 10*val + root.val);
    if(root.right !== null) ret += sumTree(root.right, 10*val + root.val);
    return ret;
}